quodlibet/quodlibet/ext/events/waveformseekbar.py

```diff
@@ -215,11 +222,15 @@ def draw_waveform(self, cr, width, height, elapsed_color, remaining_color):
             u1 = max(0, int(floor((x - hw) * ratio_width)))
             u2 = min(int(ceil((x + hw) * ratio_width)), len(data))
             val = (sum(data[u1:u2]) / (ratio_height * (u2 - u1))
-                   if u1 != u2 else 0.5)
+                   if u1 != u2 else 0.0)
 
-            # Draw single line, ensuring 1px minimum
-            cr.move_to(x, half_height - val)
-            cr.line_to(x, ceil(half_height + val))
```

Comment:
The thing is that doing a 'ceil' draws a line that is *at least* one pixel high. Say val=.25, you will draw a line of 1.25px high.

(BTW I highly suspect the reason of the anti-aliasing disabling in the code has been made for this reason. I think it could be removed once the drawing code has been cleaned.)

Not even to speak that if you have a resolution higher than the coordinate system (HiDPI, printing, ...), then using 'ceil' completely fails !